### ISSUE TITLE
Crash when discarding settings rapidly

### ISSUE THREAD

**Issue by mimvdb** [microsoft/terminal#12333]
### Windows Terminal version

1.12.3472.0

### Windows build number

10.0.19043.0

### Steps to reproduce

Start fresh window, go to settings
Rapidly click on save, then rapidly click on discard in the settings UI

To reliable reproduce this I used an autohotkey script I had lying around to click the leftmousebutton very rapidly, this may be useful (Usage: Shift+F12, then after first left click starts clicking very rapidly, press F12 to stop)
```
#NoEnv  ; Recommended for performance and compatibility with future AutoHotkey releases.
; #Warn  ; Enable warnings to assist with detecting common errors.
SendMode Input  ; Recommended for new scripts due to its superior speed and reliability.
SetWorkingDir %A_ScriptDir%  ; Ensures a consistent starting directory.

Suspend, on

F12::Reload
+F12::Suspend, off

Lbutton::

Loop
{
;GetKeyState, state, Lbutton, P
;if state=U
;break
Sendinput {Click down left}
Sleep 10
Sendinput {Click up left}
Sleep 10
}
```

### Expected Behavior

Nothing much

### Actual Behavior

Crash 50% of the time, freeze in other cases

https://user-images.githubusercontent.com/2277504/152193847-1e44455d-1bb7-4a4c-af01-8d1e463a488f.mp4

**Comment by zadjii-msft** [comment]
Interesting. I can repro this crash on `main`, but I can't repro it on the #14630 branch. However, the stack I'm seeing here is fully different than the #13673 stack, so maybe it's coincidence. 

Assuming that PR merges for 1.17, there's a chance this will be fixed in 1.17.

**Comment by zadjii-msft** [comment]
I can't get this to repro anymore on 1.18. I bet this _was_ fixed after all.
